packages/core/elasticsearch/core-elasticsearch-client-server-internal/src/cluster_client.ts

```diff
@@ -133,6 +137,7 @@ export class ClusterClient implements ICustomClusterClient {
 
     return {
       ...DEFAULT_HEADERS,
+      ...this.userAgentHeaders,
```

Comment:
This will only add the header for a "scoped" client, which in Kibana means a client that uses the credentials of the current request. Plugins can also use the `asInternalUser` public property on this class. So I think we will have to inject the headers in the `configureClient` function in order to add them to both clients.

Reply:
> So I think we will have to inject the headers in the configureClient function in order to add them to both clients.

This is right. I thought I had a draft PR somewhere, but apparently I trashed my local branch before pushing, so...

We want the change to be done as low as within the `parseClientOptions` method, which means passing the Kibana version / Env info though `ClusterClient` and `configureClient` down to `parseClientOptions`